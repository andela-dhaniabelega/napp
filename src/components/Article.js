import React from 'react';
import { render } from 'react-dom';
//import 'react-router';
import Request from 'superagent';
import _ from 'lodash';
import '../../sass/styles.scss';
import * as newsActions from '../actions/newsActions';
import newsStore from '../stores/newsStore';
import queryString from 'query-string';




export default class Article extends React.Component {

	constructor (){
		super();
		this.state = {
			searchString : '',
			articles: []
		};

		this.fetchNewsArticles = this.fetchNewsArticles.bind(this);
	}

	fetchNewsArticles(){
		this.setState({ articles: newsStore.fetchNewsArticles() });
	}


	componentWillMount(){
		const parsed = queryString.parse(this.props.location.search);
		var sourceName = parsed.sourceId;

		newsActions.getArticles(sourceName);
		newsStore.on('articles_change',this.fetchNewsArticles);

		
	}

	
	render() {
		
		var articles = _.map(this.state.articles);

		

		return (
			<div>
			

			<ul> 

                { articles.map(function(articleName,index){
                   return <li key={index}>{articleName.aurthor} {articleName.title} </li>
                }) }

            </ul>
            </div>


		);


	}

	
}	