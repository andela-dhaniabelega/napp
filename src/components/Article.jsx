import queryString from 'query-string';
import React from 'react';
import _ from 'lodash';
import '../../sass/styles.scss';
import * as newsActions from '../actions/newsActions';
import newsStore from '../stores/newsStore';
import Header from '../components/Header';
import Footer from '../components/Footer';


export default class Article extends React.Component {

	constructor (){
		super();
		this.state = {
			searchString : '',
			articles: []
		};

		this.fetchNewsArticles = this.fetchNewsArticles.bind(this);
		
	}

	componentWillMount(){
		window.parsed = queryString.parse(location.search);
		let sourceName = window.parsed.sourceId;
    let sortType = window.parsed.sortOptions;
		newsActions.getArticles(sourceName,sortType);
		newsStore.on('articles_change',this.fetchNewsArticles);

	}


	fetchNewsArticles(){
		this.setState({ articles: newsStore.fetchNewsArticles() });
	}

	render() {
		
		let articles = _.map(this.state.articles);

      return (
        <div className="wrapper">
          <div className="center">
            <Header />
            <div className="main_content2 floatleft">
              <div className="left_coloum2 floatleft">
                <br /><br />
                <ul> 
                  <div className="row">     		
                    { articles.map(function(articleName){
                      return (
                        <div className="col-lg-6">
                          <h3 className="title article-title">
                            {articleName.title}
                          </h3>
                          <img src={articleName.urlToImage} alt="" />
                          <h3 className="article-author">
                            by {articleName.author}
                          </h3>
                          <p>{articleName.description}</p>
                          <a className="readmore" href={articleName.url}>
                            read more
                          </a>
                        </div>
                      );
                    }) 
                }
                  </div>
                </ul>
              </div>


              <div className="banner0-box">
                <div className="banner0">
                  <div className="google">
                    <h2>Enjoying Napp?</h2>
                    <p>
                      Tweet at us using <a>@thenappjournal</a>
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <br /><br />
            <Footer />
          </div>
        </div>
		);
	}
	
}	