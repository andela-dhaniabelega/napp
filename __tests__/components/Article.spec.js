import React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Article from '../../src/components/Article';

const shallowRenderer = new ReactShallowRenderer();
shallowRenderer.render(<Article />);

describe('Article Page', () => {
	describe('Tests for JSX', () => {
		it('renders a snapshot', () => {
          const tree = renderer.create(<Article />).toJSON();
          expect(tree).toMatchSnapshot();
		});
    });

    describe('Render function', () => {
		it('Renders the root elements', () => {
			const renderedRoot = shallowRenderer.getRenderOutput();
			expect(renderedRoot.type).toEqual("div");
			expect(renderedRoot.props.className).toEqual('wrapper');    
		});
    });

    describe('Interactions', () => {
        const wrapper = mount(<Article />)
		it('should set state correctly (Articles)', () => {
			wrapper.setState({
				articles: [{
					urlToImage: 'urlToImage',
					title: 'Article Title',
					description: 'Article Description',
					author: 'Article Author'
				}]
           }); 
           const h3_title = wrapper.find('.truncate');
           const h3_author = wrapper.find('.article-author');
           const p = wrapper.find('.article-description');
           expect(h3_title.text()).toBe('Article Title');
           expect(h3_author.text()).toBe('by Article Author');  
           expect(p.text()).toBe('Article Description');    
		});

		it('should set state correctly (Sources)', () => {
			wrapper.setState({
				sources: [{
					name: 'Source Name',
					sortBysAvailable:[
						'top',
						'latest'
					]
				}]
           }); 
           const h2 = wrapper.find('.source-name'); 
           expect(h2.text()).toBe('Source Name');  
		});
    });


    
});