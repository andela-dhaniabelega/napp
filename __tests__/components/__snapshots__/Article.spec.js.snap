// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Page Tests for JSX renders a snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="center"
  >
    <div
      className="header"
    >
      <div
        className="logo floatleft"
      >
        <a
          href="/"
        >
          <img
            alt=""
            src="../public/images/logo copy.png"
          />
        </a>
      </div>
      <div
        className="top_menu floatleft"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About Napp
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Contact us
            </a>
          </li>
          <li>
            <a
              onClick={[Function]}
              role="button"
            >
              Sign In with Google
            </a>
          </li>
        </ul>
      </div>
    </div>
    <ul />
    <button
      className="btn btn-info backer"
      onClick={[Function]}
    >
      Go back
    </button>
    <div>
      <div
        className="main_content2 floatleft"
      >
        <div
          className="left_coloum2 floatleft"
        >
          <br />
          <br />
          <ul>
            <div
              className="row"
            />
          </ul>
        </div>
        <div
          className="banner0-box"
        >
          <div
            className="banner0"
          >
            <div
              className="google"
            >
              <h2>
                Enjoying Napp?
              </h2>
              <p>
                Tweet at us using 
                <a>
                  @thenappjournal
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
      <br />
      <br />
    </div>
    <div
      className="footer_bottom_area"
    >
      <div
        className="copyright_text"
      >
        <p>
          Copyright Â© 2017 Napp. All rights reserved
        </p>
      </div>
    </div>
  </div>
</div>
`;
